app/javascript/components/patient/close_contacts/CloseContactModal.js

```diff
@@ -0,0 +1,253 @@
+  constructor(props) {
+    super(props);
+    this.state = {
+      first_name: props.currentCloseContact.first_name,
+      last_name: props.currentCloseContact.last_name,
+      primary_telephone: props.currentCloseContact.primary_telephone,
+      email: props.currentCloseContact.email,
+      last_date_of_exposure: props.currentCloseContact.last_date_of_exposure,
+      assigned_user: props.currentCloseContact.assigned_user,
+      contact_attempts: props.currentCloseContact.contact_attempts,
+      notes: props.currentCloseContact.notes || '',
+      errors: {},
+      isValid:
+        (props.currentCloseContact.first_name || props.currentCloseContact.last_name) &&
+        (props.currentCloseContact.primary_telephone || props.currentCloseContact.email),
+    };
+  }
+
+  validateAndSubmit = () => {
+    schema
+      .validate({ ...this.state }, { abortEarly: false })
+      .then(() => {
+        this.props.onSave(this.state);
+      })
+      .catch(err => {
+        // Validation errors, update state to display to user
+        if (err && err.inner) {
+          let issues = {};
+          for (const issue of err.inner) {
+            issues[issue['path']] = issue['errors'];
+          }
+          this.setState({ errors: issues });
+        }
+      });
+  };
+
+  handleDateChange = event => this.setState({ last_date_of_exposure: event });
+
+  handleChange = event => {
+    let stateKey = event.target.id;
+    if (event?.target?.value && typeof event.target.value === 'string' && event.target.value.match(/^\s*$/) !== null) {
+      // Empty spaces are allowed to be typed (for example, a first name may be 'Mary Beth')
+      // But empty starting first spaces should not be allowed
+      event.target.value = '';
+    }
+    let value;
+    if (event?.target?.id && event.target.id === 'cc_assigned_user') {
+      stateKey = 'assigned_user';
+      if (isNaN(event.target.value) || parseInt(event.target.value) > 999999) return;
+      // trim() call included since there is a bug with yup validation for numbers that allows whitespace entry
+      value = _.trim(event.target.value) === '' ? null : parseInt(event.target.value);
+    } else if (event?.target?.id && event.target.id === 'primary_telephone') {
+      value = event.target.value.replace(/-/g, '');
+    } else {
+      value = event.target.value;
+    }
+    this.setState({ [stateKey]: value }, () => {
+      let isValid = (this.state.first_name || this.state.last_name) && (this.state.primary_telephone || this.state.email);
+      this.setState({
+        isValid,
+      });
+    });
+  };
+
+  render() {
+    return (
+      <Modal size="lg" show centered onHide={this.props.onClose}>
+        <h1 className="sr-only">{this.props.title}</h1>
+        <Modal.Header>
+          <Modal.Title>{this.props.title}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body className="px-5">
+          <Row className="mt-3">
+            <Form.Group as={Col} lg="12" controlId="first_name">
```

Comment:
thoughts on changing all the ids to `cc_[field]` and then just stripping the `cc_` off in the onChange? might make this file more consistent

... yes I know I am the one that originally did this ...

Reply:
not the worst idea. I'll take a look

Reply:
was a good idea. done